lib/Serialization/ModuleFile.h

```diff
@@ -888,8 +896,15 @@ class ModuleFile
   /// Returns the decl with the given ID, deserializing it if needed.
   ///
   /// \param DID The ID for the decl within this module.
+  ///
+  /// \param matchAttributes Optional check on the attributes of the decl to
+  /// determine if it should be fully deserialized and returned. If the
+  /// attributes fail the check, the decl is not deserialized and
+  /// \c DeclAttributesDidNotMatch is returned.
   llvm::Expected<Decl *>
-  getDeclChecked(serialization::DeclID DID);
+  getDeclChecked(
+    serialization::DeclID DID,
+    llvm::function_ref<bool(DeclAttributes)> matchAttributes = nullptr);
```

Comment:
Do we need to pass down this to the function call? can we check at the caller side?

Reply:
To add to my answer above, the function is used in the middle of the deserialization operation, after deserializing the attributes and before the full Decl.